resources/tag_models/event.json

```diff
@@ -0,0 +1,29 @@
+{
+  "tag": {
+	"api_key": "", // String - API key - Make sure the request is from a valid client (idea nicked from Keen.io) useful if a page gets copied onto a Russian website and creates noise
+	"version": "", // Version of the tracking client e.g. "JS v1.2"
+	"type": "event", // Type of request (page or event)
+	"id": "", // ID of this request, must be unique
+	"offset": 0 // Delay of this tag between event happening and being sent to server - milliseconds
+  },
+
+  "event": {
+	"page_id": "", // ID of the page on which this event happened.
+	"parent_id": "", // Parent event ID if this is a child event.
```

Comment:
On understanding: Why don't you do a survey?  My vote is that your parent-child event model is extremely counter-intuitive. If everyone else thinks it makes perfect sense, then there's no problem here.  YOU understand your model extremely well.  The question is do you want others to understand it quickly and easily and therefore instinctively send good data without training, or do you want to battle with dodgy data coming in as a result of misunderstanding the model, and run regular training courses to teach people how the model works?

On implementation: the simple issue is that we want no shared state.  It's not loading o-tracking that I think will cause leaks, it's maintaining a ton of shared state everywhere so that components know what their 'parent event' is.  Any solution that avoids shared state is fine from a component development perspective.  So for example, if you want to organise events in your backend like this:

```
Page session
 \-- Video
  |-- Player 1
  \-- Player 2
    \-- Clip 1
      |-- Play
      \-- Scrub
```

Then to me, none of these are events except for Play and Scrub.  The others are all aggregations/collations of those events.  Now, if you accept events with a payload like this:

```
{
  collations: [
    "videoclip:12345",
    "o-video-instance:0000-0000-000-0000",
    "origami:o-video"
  ]
}
```

Then when that event arrives at o-tracking, you could add some higher level collations, like the page session ID, and then consider that sequence to be a hierarchy, if that is best for your model.

Incidentally, I'm no expert in data modelling and information retrieval, but I seem to remember something about sparse table algorithms that would potentially enable you to do this without a massive n:m join.  The most common collation types that you might want to query could be promoted to first-class event properties, like 'page session ID', since o-tracking is likely to add that to every single event.

If if you do opt to use a hierarchical model you should be able to convert it later into a flat collation system simply by treating every ancestor of an event as one of its collation IDs.